app/Policies/UserPolicy.php

```diff
@@ -122,7 +128,22 @@ public function attachTeam(User $user, User $userResource, Team $team)
      */
     public function attachAnyTeam(User $user, User $userResource)
     {
-        return $user->can('update-teams-membership');
+        if ($user->can('update-teams-membership')) {
+            return true;
+        }
+        if ($user->can('update-teams-membership-own') && $user->is($userResource)) {
+            return true;
+        }
+        $user_manages = $user->manages()->get();
+        if (count($user_manages) > 0) {
```

Comment:
I don't think this logic works?

> This method will return the values in the original collection that are not present in the given collection

For example, user manages Outreach and is a member of Outreach and Battlebots. The attach command is called for Battlebots.
```
$user_manages = ['Outreach'];
$target_in = ['Outreach', 'Battlebots'];
$diff = ['Battlebots'];
```
Since there's a difference, it'll allow it which I don't think is the correct behavior since the user doesn't manage Battlebots. That is, unless I have something backwards.

Reply:
Battlebots isn't in 'the original collection'